src/rust/zh.baozimanhua/src/parser.rs

```diff
@@ -95,74 +92,76 @@ pub fn get_filtered_url(filters: Vec<Filter>, page: i32, url: &mut String) {
 		query.push("q", Some(search_str.as_str()));
 
 		url.push_str(format!("/search?{}", query).as_str());
-	} else {
-		query.push("type", Some(c_type));
-		query.push("region", Some(c_region));
-		query.push("filter", Some(c_filter));
-		query.push("page", Some(page.to_string().as_str()));
-		query.push("limit", Some("20"));
-		query.push("language", Some("tw"));
-
-		url.push_str(format!("{}?{}", API_URL, query).as_str());
+
+		return url;
 	}
+	query.push("type", Some(classify_type));
+	query.push("region", Some(classify_region));
+	query.push("filter", Some(classify_filter));
+	query.push("page", Some(page.to_string().as_str()));
+	query.push("limit", Some("20"));
+	query.push("language", Some("tw"));
+
+	url.push_str(format!("{}?{}", API_URL, query).as_str());
+
+	url
 }
 
 pub fn request_get(url: String) -> Request {
-	Request::new(url, HttpMethod::Get).header("User-Agent", USER_AGENT)
+	Request::get(url).header("User-Agent", USER_AGENT)
 }
 
-pub fn parse_home_page(json_data: ValueRef) -> Result<MangaPageResult> {
-	let mut mangas: Vec<Manga> = Vec::new();
-
-	let mut has_more = false;
+pub fn parse_home_page(json: ValueRef) -> Result<MangaPageResult> {
+	let object = json.as_object()?;
 
-	let object = json_data.as_object()?;
-	if object.len() != 1 {
-		has_more = true;
+	if object.len() == 1 {
+		return Ok(MangaPageResult {
+			..Default::default()
```

Comment:
you could just do `MangaPageResult::default()`, but I think this is a little unclear honestly, so I would do an explicit init with `has_more: false`, but honestly I just need to update aidoku-rs so don't worry about it